packages/components/src/primitives/svg/index.native.js

```diff
@@ -21,24 +19,32 @@ export const SVG = ( props ) => {
 	// We're using the react-native-classname-to-style plugin, so when a `className` prop is passed it gets converted to `style` here.
 	// Given it carries a string (as it was originally className) but an object is expected for `style`,
 	// we need to check whether `style` exists and is a string, and convert it to an object
-	let styleKeys = new Array();
-	const styleValues = new Array();
-	if ( typeof props.style === 'string' || props.style instanceof String ) {
-		styleKeys = props.style.split( ' ' );
-		styleKeys.forEach( ( element ) => {
-			const oneStyle = styles[ element ];
-			if ( oneStyle !== undefined ) {
-				styleValues.push( oneStyle );
-			}
-		} );
+
+	let styleValues = {};
+	if ( typeof props.style === 'string' ) {
+		let oneStyle = props.style.split( ' ' ).map( element => styles[ element ] ).filter( Boolean );
+		styleValues = Object.assign( styleValues, ...oneStyle );
 	}
 
-	const safeProps = styleValues.length === 0 ? { ...omit( props, [ 'style' ] ) } : { ...props, style: styleValues };
+	const safeProps = { ...props, style: styleValues };
+
 	return (
 		<Svg
+			//We want to re-render when style color is change
+			key={safeProps.style.color}
 			height="100%"
 			width="100%"
 			{ ...safeProps }
 		/>
 	);
 };
+
+export const Path = ( props ) => {
```

Comment:
I don't understand why we need this wrapper, can you explain more?

Reply:
Good catch! This was a valid solution at some point, but after a few iterations, it lost purpose!
Fixed.